File: backend/digitalhouse-pi/src/main/java/com/pi/dh/dto/CursaDisciplinaOferecidaDTO.java
package com.pi.dh.dto;

import java.io.Serializable;
import java.time.OffsetDateTime;
import java.util.List;

import lombok.Data;

@Data
public class CursaDisciplinaOferecidaDTO implements Serializable {
	
	private static final long serialVersionUID = 1L;

	private Long cursaDisciplinaOferecidaId;
	private OffsetDateTime dataInscricao;
	private OffsetDateTime dataDeferimento;
	private Integer faltas;
	private AlunoDTO aluno;
	private DisciplinaOferecidaDTO disciplinaOferecida;
	private List<NotaDTO> notas;

}

File: frontend/sistema-academico/src/app/shared/dto/cursa-disciplina-oferecida.dto.ts
import { DisciplinaOferecidaDto } from './disciplina-oferecida.dto';
import { AlunoDto } from './aluno.dto';
import { NotaDto } from './nota.dto';

export interface CursaDisciplinaOferecidaDto {
  cursaDisciplinaOferecidaId?: number,
  dataInscricao?: Date,
  dataDeferimento?: Date,
  faltas?: Number,
  aluno?: AlunoDto,
  disciplinaOferecida?: DisciplinaOferecidaDto,
  notas?: NotaDto
}